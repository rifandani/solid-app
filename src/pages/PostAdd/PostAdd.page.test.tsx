import { Route } from '@solidjs/router';
import { waitFor } from '@solidjs/testing-library';
import { renderWithRouter } from '../../utils/test.util';
import PostAddPage from './PostAdd.page';

describe('PostAddPage', () => {
  it('should render correctly', async () => {
    // ARRANGE
    const { container, getByText } = renderWithRouter(() => (
      <Route path="/" component={PostAddPage} /> // the path doesn't matter here
    ));

    // ASSERT
    await waitFor(() => {
      expect(getByText(/Add Post/)).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
