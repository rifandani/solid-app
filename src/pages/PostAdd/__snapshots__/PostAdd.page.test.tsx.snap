// Vitest Snapshot v1

exports[`PostAddPage > should render correctly 1`] = `
<main
  class="py-20 px-10 md:px-24 lg:px-40 xl:px-52 flex flex-col items-center justify-center"
>
  <h1
    class="text-2xl font-semibold tracking-wider text-violet-500"
  >
    Add Post
  </h1>
  <form
    class="flex flex-col gap-3 w-full"
  >
    <label
      for="title"
    >
      Title
    </label>
    <input
      autofocus=""
      class="mt-1"
      name="title"
      placeholder="Title..."
      required=""
      type="text"
    />
    <label
      for="body"
    >
      Body
    </label>
    <textarea
      class="mt-1"
      name="body"
      placeholder="Body..."
      required=""
      rows="3"
    />
    
    <button
      class="my-5"
      type="submit"
    >
      Submit
    </button>
  </form>
</main>
`;
